pkg/storage/mysql/mysqlstore.go

```diff
@@ -217,7 +217,72 @@ type sqlDBResultsIterator struct {
 }
 
 func (s *sqlDBStore) CreateIndex(createIndexRequest storage.CreateIndexRequest) error {
-	// TODO: Issue-36 Implement Create Index
+	// get all the created indexes
+	indexes, err := s.getIndexes()
+	if err != nil {
+		return fmt.Errorf("failed to get indexes %s", err)
+	}
+	// if an index exits, drop it as sql throws duplicate key_name error
+	err = s.dropExistingIndex(indexes, createIndexRequest)
+	if err != nil {
+		return fmt.Errorf("failed to drop an existing index %s", err)
+	}
+	// create an index
+	createIndexStmt := "CREATE INDEX " + createIndexRequest.IndexName + " ON " +
+		createIndexRequest.IndexStorageLocation + " (" + createIndexRequest.WhatToIndex + ")"
+
+	_, err = s.db.Exec(createIndexStmt)
+	if err != nil {
+		return fmt.Errorf("failed to create index %w", err)
+	}
+
+	return nil
+}
+
+func (s *sqlDBStore) getIndexes() ([]string, error) {
+	var index string
+
+	var indexes []string
+
+	getIndexStmt := "SELECT DISTINCT INDEX_NAME FROM INFORMATION_SCHEMA.STATISTICS" +
+		" WHERE TABLE_SCHEMA = ?"
+
+	indexStmt, err := s.db.Prepare(getIndexStmt)
+	if err != nil {
+		return nil, fmt.Errorf("failed to prepare select index statement %w", err)
+	}
+
+	rows, err := indexStmt.Query(s.tableName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to query select index statement %w", err)
+	}
+
+	// A table by default have clustered index named as PRIMARY and can contain more than one index
```

Comment:
Grammar: should be either

`// A table by default has a clustered index named as PRIMARY and can contain more than one index`
or
`// Tables by default have a clustered index named as PRIMARY and can contain more than one index`